src/ioctls/system.rs

```diff
@@ -374,6 +376,46 @@ impl Kvm {
         }
     }
 
+    /// AArch64 specific function to create a VM fd using the KVM fd with flexible IPA size.
+    ///
+    /// See the arm64 section of KVM documentation for `KVM_CREATE_VM`.
+    /// A call to this function will also initialize the size of the vcpu mmap area using the
+    /// `KVM_GET_VCPU_MMAP_SIZE` ioctl.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use kvm_ioctls::{Kvm, Cap};
+    /// let kvm = Kvm::new().unwrap();
+    /// // Check if the ArmVmIPASize cap is supported.
+    /// if kvm.check_extension(Cap::ArmVmIPASize) {
+    ///     let vm = kvm.create_vm_with_ipa_size(40).unwrap();
```

Comment:
Should we use another value here? It looks like 40 is the default.

Reply:
hmmm yes. Actually the reason why I used this default value is that I found it very difficult to choose a proper value, as different arm servers might have a different implementation of the `Host_IPA_Size`. Therefore the guest IPA is limited. 

The register [TCR_EL2](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0488c/CIHJBCHI.html) and [VTCR_EL2](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0488d/CIHEDIII.html) indicate that the possible values are 32, 36, 40, 42, 44, 48. Any suggestions about this?

Reply:
Hm, let's stick with 40 then, but we should add a comment that this is the default value. I think it would be useful to also say one line about the registers as well.

Reply:
Thanks @andreeaflorescu, all done. Please take a look.

Reply:
Why not test with the maximum that the host supports? That would both test the discoverability of supported IPA sizes and it would likely test with something other than 40.

Reply:
Yeah that makes sense...@andreeaflorescu what do you think? I can change to this way.

Reply:
Yap, great idea!

Reply:
Thanks for the suggestion @rhdrjones , I updated the code and would you mind taking another look?

@andreeaflorescu Sorry to keep you busy but I think you have to approve again :P

Reply:
wrt the test below, it looks good to me. Thanks

Reply:
> @andreeaflorescu Sorry to keep you busy but I think you have to approve again :P

Don't worry about it. It's my job to maintain this crate. When I don't have time, I might end up ghosting people (by mistake) :))

Reply:
@MrXinWang should we also update the code example to have the ipa_size = host_ipa_size?

Reply:
Done @andreeaflorescu. Thought in the code example we can have any values instead of being as strict as in the unit test.

Reply:
The code examples are also run in the CI, but it should be fine.